example/lib/main.dart

```diff
@@ -68,14 +84,83 @@ class _MyAppState extends State<MyApp> {
     );
   }
 
+  Future<void> initPlatformState() async {
+    String uid;
+    try {
+      uid = await Qonversion.launch('PV77YHL7qnGvsdmpTs7gimsxUvY-Znl2');
+      print(uid);
+    } catch (e) {
+      print('Failed to obtain uid from Qonversion.');
+      print(e);
+    }
+
+    if (!mounted) return;
+
+    setState(() => _uid = uid);
+
+    final purchaseUpdates =
+        InAppPurchaseConnection.instance.purchaseUpdatedStream;
+    _subscription = purchaseUpdates.listen((purchases) {
+      _handlePurchaseUpdates(purchases);
+    });
+
+    final available = await InAppPurchaseConnection.instance.isAvailable();
+    if (!available) {
+      print('Store is not available');
+    }
+
+    final _kIds = Set<String>.from(['qonversion_subs_monthly']);
+    final response =
+        await InAppPurchaseConnection.instance.queryProductDetails(_kIds);
+
+    print(response.productDetails);
+
+    setState(() => _products = response.productDetails);
+  }
+
+  void _handlePurchaseUpdates(List<PurchaseDetails> purchases) async {
+    for (var i = 0; i < purchases.length; i++) {
+      if (purchases[i].status == PurchaseStatus.error) {
+        print('Bad purchase status');
+        continue;
+      }
+
+      if (!purchases[i].billingClientPurchase.isAcknowledged) {
+        print('Going to acknowledge purchase ${purchases[i].productID}');
+        await InAppPurchaseConnection.instance.completePurchase(purchases[i]);
+        print('Successfully acknowledged purchase ${purchases[i].productID}');
+      }
+
+      final uid =
+          await Qonversion.manualTrackPurchase(_products[0], purchases[0]);
+      print('Qonversion uid: $uid');
+    }
+  }
+
+  void makePurchase() async {
+    if (_mockTrack) {
+      final uid = await Qonversion.manualTrackPurchase(
+          getProductDetails(), getPurchaseDetails());
+      print(uid);
+      return;
+    }
+    final productDetails = _products[0];
+    final purchaseParam = PurchaseParam(productDetails: productDetails);
+
+    final res = await InAppPurchaseConnection.instance
+        .buyNonConsumable(purchaseParam: purchaseParam);
+
+    print('Item purchased, result: $res');
+  }
+
   ProductDetails getProductDetails() {
     // ignore: invalid_use_of_visible_for_testing_member
     final original = SkuDetailsWrapper(
       description: 'description',
       freeTrialPeriod: 'freeTrialPeriod',
       introductoryPrice: 'introductoryPrice',
       introductoryPriceMicros: 'introductoryPriceMicros',
-      introductoryPriceCycles: 'introductoryPriceCycles',
+      introductoryPriceCycles: null,
```

Comment:
Are you sure this correct?
That is how it works in Android SDK
`return if (details.freeTrialPeriod.isNullOrEmpty()) details.introductoryPriceCycles else 0`

Reply:
that's just a mock to test if parsing null values works correctly. doesn't make any difference here

Reply:
Oh, my bad, sorry 🙃